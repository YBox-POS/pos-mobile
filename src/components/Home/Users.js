import React from 'react';
import { Text, ScrollView, View, Button, StyleSheet } from 'react-native';
import styled from 'styled-components/native';
import { List, ListItem } from 'react-native-elements'


const Users = ({ users, followUser, unfollowUser }) => (
  <ScrollView
    style={{
      flex: 1
    }}
    contentContainerStyle={{ flex: 1 }}
  >
    {Object.values(users).map(({ id, name, following }) => (
      <ListItem
        title={name}
        key={`user-${id}`}
        rightIcon={
          <Button
            color={"tomato"}
            onPress={() => following ? unfollowUser(id) : followUser(id)}
            title={following ? '取消关注' : '关注'}
          /> 
        }
      />
    ))}
  </ScrollView>
)

export default Users;
