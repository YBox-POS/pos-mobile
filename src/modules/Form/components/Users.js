import React from 'react';
import { Text, Image, View } from 'react-native';
import GridView from 'react-native-super-grid';
import styles from './styles';


const ExImages = ({ imagesa, followImage, unfollowImage }) => (

  <GridView
    itemWidth={103}
    items={imagesa}
    style={styles.gridView}
    renderItem={item => (
      <View style={styles.itemContainer}>          
        <Image
          style={styles.imagesize} 
          source={{url:item.path}}>
          {/* <Text >这是嵌在图中的一句话</Text> */}
        </Image>
          <View style={styles.testview}>
            <Text style={styles.itemName}>{item.name}</Text>
          </View>
      </View>
    )}
  />   
 
)



export default ExImages;
