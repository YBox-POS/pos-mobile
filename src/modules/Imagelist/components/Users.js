import React from 'react';
import { Text, ScrollView, Image, View, StyleSheet } from 'react-native';
import styled from 'styled-components/native';
import {ListItem, Button} from 'react-native-elements';
import GridView from 'react-native-super-grid';
import styles from './styles';

const ExImages = ({ imagesa, followImage, unfollowImage }) => (
  <ScrollView
    style={{
      flex: 1,
    }}
  >
    <Text>{console.log("|||||||||||||||||."+ Object.prototype.toString.call(imagesa).slice(8,-1))}</Text>  
     {/* <Text>{console.log("//////////"+imagesa)}</Text>  */}
    {/* {Object.values(imagesa).map(({ id, path, price, name,desc }) => (  */}
        <GridView
        itemWidth={120}
        items={imagesa}
        style={styles.gridView}
        renderItem={item => (
          <View style={[styles.itemContainer, { backgroundColor: item.code }]}>
            <image src ={item.path}/>
            <Text style={styles.itemName}>{item.name}</Text>
            <Text style={styles.itemCode}>{item.desc}</Text>
          </View>
        )}
      />      
       {/* ))}     */}
     {/* {Object.values(imagesa).map(({ id, path, price, name,desc }) => (
      <ListItem
        title={name}
        avatar={{uri:path}}
        key={`image-${id}`}
          rightIcon={
          <Button
            color={"#fff"}
            iconRight
            buttonStyle={{backgroundColor: '#00D9C7', borderRadius: 5}}
            title={'购买'}
          /> 
        }
      />
    ))}  */}
  </ScrollView>
)



export default ExImages;
