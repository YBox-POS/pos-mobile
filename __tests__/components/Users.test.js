// __tests__/Intro-test.js
import 'react-native';
import React from 'react';
import Users from '../../src/components/Home/Users';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Users
      users={
        [
          {id: 1, name: 'Test 1', following: true},
          {id: 2, name: 'Test 2', following: false},
          {id: 3, name: 'Test 3', following: true}
        ]
      }

    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});