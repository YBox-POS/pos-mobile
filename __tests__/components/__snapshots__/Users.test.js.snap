// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLongPress={undefined}
    onPress={undefined}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "#ededed",
          "borderBottomWidth": 1,
          "paddingBottom": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
    underlayColor="white"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginLeft": 10,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 16.099999999999998,
                  },
                  Object {
                    "marginLeft": 10,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Test 1
          </Text>
        </View>
        <View
          style={undefined}
        >
          <View />
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "tomato",
                },
              ]
            }
          >
            取消关注
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    onLongPress={undefined}
    onPress={undefined}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "#ededed",
          "borderBottomWidth": 1,
          "paddingBottom": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
    underlayColor="white"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginLeft": 10,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 16.099999999999998,
                  },
                  Object {
                    "marginLeft": 10,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Test 2
          </Text>
        </View>
        <View
          style={undefined}
        >
          <View />
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "tomato",
                },
              ]
            }
          >
            关注
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    onLongPress={undefined}
    onPress={undefined}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "#ededed",
          "borderBottomWidth": 1,
          "paddingBottom": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
    underlayColor="white"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginLeft": 10,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 16.099999999999998,
                  },
                  Object {
                    "marginLeft": 10,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Test 3
          </Text>
        </View>
        <View
          style={undefined}
        >
          <View />
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "tomato",
                },
              ]
            }
          >
            取消关注
          </Text>
        </View>
      </View>
    </View>
  </View>
</ScrollView>
`;
